### PR TITLE
All Errors handled by file validators

### PR DESCRIPTION
**Pull request by somewhatabstract**
This changes the `generateMarkerEdges` function to treat missing
target files or markers as broken edges rather than reporting them
directly. The validators have then been updated to understand this
kind of broken edge and report them. This will allow for other kinds
of validators to handle this kind of error with a more helpful message
(such as those that could be used to support linters).

This also addresses an issue that Flow can't catch due to poor core
typing. It turns out that regex optional groups can come back as
undefined in a match, even though the type is a string. So, now
the two optional groups in our tag parsing are coerced to empty strings
in the case that they aren't present (which is what was expected).

### CODE REVIEW

File: src/marker-parser.js [hunk 1]
```diff
@@ -328,13 +337,16 @@ export default class MarkerParser {
...
-                    startDecode[1],
-                    startDecode[3],
+                    startDecode[TagDecodeGroup.tagId],
+                    startDecode[TagDecodeGroup.targetFileName],
                     lineNumber,
-                    startDecode[2],
+                    startDecode[TagDecodeGroup.checksum] || "",
                     startMatch[1],
-                    startDecode[4],
+                    startDecode[TagDecodeGroup.commentEnd] || "",
```
Comment (kevinbarabash) [inline src/marker-parser.js:349]: Nice way to document which group contains what.

File: src/__tests__/validate-and-fix_test.js [hunk 2]
```diff
@@ -45,6 +45,115 @@ describe("#validateAndFix", () => {
         expect(result).toBeTrue();
     });
 
+    it("should resolve false if the target of a tag does not exist", async () => {
```
Comment (kevinbarabash) [inline src/__tests__/validate-and-fix_test.js:48]: nit: I found `... where target file does not reference source file` from marker-parser_test.js to be more descriptive of this situation.

**Review by kevinbarabash — APPROVED** [review]
LGTM.  Thanks for reworking the architecture to consolidate reporting of errors and validations in the same spot. :shipit:

File: src/marker-parser.js (hunk 1 above)
Comment (somewhatabstract) [reply src/marker-parser.js:349]: Thanks. I was getting confused and thought some self-documenting code might help :)

**Pull request merged**